torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -8533,4 +8533,45 @@ def forward(self, x):
             self.assertIsNone(module.module.l1.bias.grad)
             self.assertIsNone(module.module.buffer.grad)
 
+        def _test_complex_tensor_with_ddp(self, device):
+            class ComplexModule(nn.Module):
+                def __init__(self):
+                    super(ComplexModule, self).__init__()
+                    self.p = torch.nn.Parameter(torch.view_as_complex(torch.tensor([1.0, 1], device=device)))
+
+                def forward(self, input):
+                    return (self.p * self.p)
+
+
+            cpp_builtin_hooks = [
+                None,
+                dist.BuiltinCommHookType.ALLREDUCE,
+                dist.BuiltinCommHookType.FP16_COMPRESS,
+            ]
+
+            for hook in cpp_builtin_hooks:
+
+                model = ComplexModule().to(device)
+                if device == "cpu":
+                    ddp_model = nn.parallel.DistributedDataParallel(model)
+                else:
+                    ddp_model = nn.parallel.DistributedDataParallel(model, device_ids=[self.rank])
+
+                if hook is not None:
+                    ddp_model._register_builtin_comm_hook(hook)
+
+                ddp_model(10).backward()
```

Comment:
I wonder if running a few iterations, computing loss, calling backwards and comparing to a local model's output would make for a better test.